### DISCUSSION TITLE
Automated BATCH operations

### DISCUSSION THREAD

**Discussion by DrYSG** [DaxStudio/DaxStudio#1275] [Ideas]
In discussion https://github.com/DaxStudio/DaxStudio/discussions/1274 #1274 

I talk about how I have to get around MSFT trottling the XMLA connector tor their Teams Call Quality Warehouse (only 10K rows at a time).

I have a work-a-round, where broke my EVAULATE (see below) in 1500 segments, and I do this all as a single DAX batch. (it only took 1.5 hours for 275MB of data!

***I was so impressed with DAX Studio that it could handle this with ease, THANK YOU. #**

What I did is template the time arguments (as shown below) and filled the ${startTime{} and ${endTime} via a python script and the wrote this out as a single DAX file.

From this effort, I have two requests:

1. Can you add a "RELOAD" file for TAB associated with a file? This would allow external updates of the Tab
2. When you do get to implementing #92 for automation, can you make sure that the powershell command can take the folling command parameters:
a. Specificy the Table in the PBI I am connecting to? (really nice to be to to spawn the PBI also, so I can do this fully from the command line)
b. Specify the DAX file (and have template-parameters in that file)
c. Be able to apply PowerShell parameters to that template DAX file

Probably this would result in sequential DAX requests, instead of BATCH, but I don't see how we can get around that.

Thank you.

Evaluate Template used:
```
EVALUATE
	CALCULATETABLE(
		SUMMARIZECOLUMNS(
			CQD[Second Network Name],
			CQD[Second Subnet],
			CQD[Date],
			CQD[Day Of Week],
			CQD[Hour],
			CQD[Session Type],
			CQD[Transport],
			CQD[Second Network Connection Detail],
			CQD[Total Call Count],
			CQD[P50 Jitter],
			CQD[P50 Latency],
			CQD[P50 Packet Loss Rate],
			CQD[Avg Network Jitter Max],
			CQD[Audio Poor Percentage],
			CQD[Video Poor Percentage],
			CQD[Audio Poor Call Stream Count],
			CQD[Video Poor Stream Count],
			CQD[Avg Jitter Max],
			CQD[Avg Packet Loss Rate Max],
			CQD[Avg Call Duration],
			CQD[Avg Round Trip Max]
		),
		FILTER(
			CommonFilters,
			CQD[End Time] >= dt"${StartTime}" &&
			CQD[End Time] < dt"${EndTime}"
		)
	)
```

**Comment by dgosbell** [comment]
> Can you add a "RELOAD" file for TAB associated with a file? This would allow external updates of the Tab

This should be possible - I've added #1276 to track this request. 

> a. Specificy the Table in the PBI I am connecting to? (really nice to be to to spawn the PBI also, so I can do this fully from the command line)

You cannot connect to a specific table, a connection is to a "server" and "database/model" and both of these will be parameters. You specify the table inside your DAX query.


> b. Specify the DAX file (and have template-parameters in that file)
c. Be able to apply PowerShell parameters to that template DAX file

I don't have plans to build a specific template/parameter option. But if you will be able to either pass a file or a string for the DAX query so you could either write out the file dynamically from PowerShell or generate a templated string in PowerShell and pass that in. So you will still be able to achieve the same end effect.

**Comment by dgosbell** [comment]
I've added a Powershell example to the docs here https://daxstudio.org/docs/features/command-line/powershell-examples/ showing one way of looping over a set of values and injecting those into a query
